sonar-db/src/main/java/org/sonar/db/component/ComponentDto.java

```diff
@@ -19,33 +19,88 @@
  */
 package org.sonar.db.component;
 
+import com.google.common.base.Splitter;
+import com.google.common.base.Strings;
 import java.util.Date;
+import java.util.List;
 import javax.annotation.CheckForNull;
 import javax.annotation.Nullable;
 import org.apache.commons.lang.builder.ToStringBuilder;
 import org.sonar.api.component.Component;
 import org.sonar.api.resources.Scopes;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static org.sonar.db.component.ComponentValidator.checkComponentKey;
 import static org.sonar.db.component.ComponentValidator.checkComponentName;
 
 public class ComponentDto implements Component {
 
-  public static final String MODULE_UUID_PATH_SEP = ".";
+  public static final String UUID_PATH_SEPARATOR = ".";
```

Comment:
I would move all PATH related constants and methods out of ComponentDto.
Some methods are used only in DAO, constants are use in very rare places
this is internal mechanics, this shouldn't be exposed in this class which is intended to be used everywhere